packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -100,17 +100,93 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// A controller for the content of a [Selectable].
+///
+/// A user can set the value on this controller to modify the content under
+/// a selection created by [SelectionArea] or [SelectableRegion].
+abstract class SelectedContentController<T extends Object> extends ValueNotifier<T> {
+  /// Creates a controller for the content of a [Selectable] or [SelectionHandler].
+  SelectedContentController(super.value);
+
+  /// Additional controllers to include as children.
+  ///
+  /// Children of a given controller enable more granular modification of the
+  /// selection.
+  List<SelectedContentController<Object>> children = <SelectedContentController<Object>>[];
```

Comment:
How exactly do the `children` relate to the `value` span and its children? Is `children` supposed to contain all SelectedContentControllers of all other spans in `value`'s subtree?

(Assuming I'm understanding this correctly) I wonder if we can do this in a way that maintains more of a single source of truth.

Reply:
Correct, the `children` contain all the `SelectedContentController`s that are within the `value`s subtree. For example selecting from "some" to "Bullet 3" in the `Text` widget below would result in a SelectedContentController with 3 children controllers representing the individual bullet `Text` widgets,

```dart
Text.rich(
  TextSpan(
    text: 'This is some bulleted list:\n',
    children: <InlineSpan>[
      WidgetSpan(
        child: Column(
          children: <Widget>[
            for (int i = 1; i <= 7; i += 1)
              Padding(
                padding: const EdgeInsets.only(left: 20.0),
                child: Text('• Bullet $i'),
              )
          ],
        ),
      ),
    ],
  ),
),
```

I think if we had no children controllers, it would become difficult to edit something like the bulleted text widget in the example, as a user would have to dig into the contents of a `WidgetSpan` manually using `(WidgetSpan.child as Column).children[index]` and we would still to communicate the selection ranges for each of the bullets.

Reply:
Ok I see... My main concern is the controllers getting out of sync with the spans. Can we use immutability somehow to prevent them from getting out of sync?

Reply:
I have made the `SelectedContentController.children` immutable as well now.